tsdb/aggregate.go

```diff
@@ -167,7 +167,19 @@ func (e *AggregateExecutor) execute(out chan *models.Row) {
 		}
 
 		row.Values = values
-		out <- row
+
+		// Check to see if our client disconnected, or it has been to long since
+		// we were asked for data...
+		select {
+		case out <- row:
+		case <-closing:
+			out <- &models.Row{Err: fmt.Errorf("execute was closed by caller")}
+			break
+		case <-time.After(30 * time.Second):
```

Comment:
same, I'm still not sure this won't cause problems on queries that hit multiple shards where the first shard takes more than 30s to churn through the results. @corylanou, you're sure this won't happen?